services/aztec-listener/src/svcs/poller/network-client/pool.ts

```diff
@@ -0,0 +1,51 @@
+import { AztecNode, createAztecNodeClient } from "@aztec/aztec.js";
+import { jsonStringify } from "@chicmoz-pkg/types";
+import { AZTEC_RPC_URL_POOL } from "../../../environment.js";
+import { logger } from "../../../logger.js";
+
+interface NodeConfig {
+  id: string;
+  url: string;
+  node: AztecNode;
+}
+
+let nodePool: NodeConfig[] = [];
+let currentNodeIndex = 0;
+
+// init() get a list of node urls form env variables
+export const initPool = () => {
```

Comment:
```suggestion
export const initPool = (config: AztecNodePoolConfig) => {
```

And then you can do the env-var parsing in the file calling this file instead.